metadata-ingestion/tests/integration/mode/mode_mces_golden.json

```diff
@@ -243,6 +245,77 @@
         "lastRunId": "no-run-id-provided"
     }
 },
+{
+    "entityType": "dataset",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:mode,5450544,PROD)",
+    "changeType": "UPSERT",
+    "aspectName": "datasetProperties",
+    "aspect": {
+        "json": {
+            "customProperties": {
+                "updated_at": "2024-09-02T07:40:44.046Z",
+                "last_run_id": "3535709679",
+                "data_source_id": "44763",
+                "report_imports_count": "2"
+            },
+            "externalUrl": "https://app.mode.com/acryl/reports/24f66e1701b6/details/queries/9b2f34343531",
+            "name": "Dataset 1",
+            "description": "### Source Code\n``` sql\n-- Returns first 100 rows from DATAHUB_COMMUNITY.POSTGRES_PUBLIC.COMPANY\n  SELECT \n\t\tAGE,\n\t\tID,\n\t\tNAME,\n\t\t_FIVETRAN_DELETED,\n\t\t_FIVETRAN_SYNCED\n FROM DATAHUB_COMMUNITY.POSTGRES_PUBLIC.COMPANY LIMIT 100;\n\n-- Returns first 100 rows from ETHAN_TEST_DB.PUBLIC.ACCOUNT_PHONE_NUMBER\n  SELECT \n\t\tCOMMUNICATION_ACCOUNT_ID,\n\t\tID,\n\t\tMMS_CAPABLE,\n\t\tPHONE_NUMBER,\n\t\tSMS_CAPABLE,\n\t\tSTATUS,\n\t\tSTATUS_TLM,\n\t\tTLM,\n\t\tVOICE_CAPABLE,\n\t\tWHEN_CREATED\n FROM ETHAN_TEST_DB.PUBLIC.ACCOUNT_PHONE_NUMBER LIMIT 100;\n \n \n```\n            ",
+            "tags": []
+        }
+    },
+    "systemMetadata": {
+        "lastObserved": 1638860400000,
+        "runId": "mode-test",
+        "lastRunId": "no-run-id-provided"
+    }
+},
+{
+    "entityType": "dataset",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:mode,5450544,PROD)",
+    "changeType": "UPSERT",
+    "aspectName": "subTypes",
+    "aspect": {
+        "json": {
+            "typeNames": [
+                "Dataset"
+            ]
+        }
+    },
+    "systemMetadata": {
+        "lastObserved": 1638860400000,
+        "runId": "mode-test",
+        "lastRunId": "no-run-id-provided"
+    }
+},
+{
+    "entityType": "dataset",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:mode,5450544,PROD)",
+    "changeType": "UPSERT",
+    "aspectName": "browsePathsV2",
+    "aspect": {
+        "json": {
+            "path": [
+                {
+                    "id": "acryl"
+                },
+                {
+                    "id": "urn:li:container:800cfcb4cec6ad587cafde11a0b0bb4a",
+                    "urn": "urn:li:container:800cfcb4cec6ad587cafde11a0b0bb4a"
+                },
+                {
+                    "id": "urn:li:dashboard:(mode,5450544)",
+                    "urn": "urn:li:dashboard:(mode,5450544)"
```

Comment:
should mode datasets be nested under a dashboard in the hierarchy? what happens if a single dataset is used by multiple reports?

imo they should be nested under the space in the browse hierarchy

Reply:
should mode datasets be nested under a dashboard in the hierarchy ? In Mode we can see reports and mode-datasets into same hierarchy level thats the reason i have added into dashboard

what happens if a single dataset is used by multiple reports? 
On Lineage level if we select Mode-datasets it will show lineage to all reports 
For eg. 
Below **NewDatasetTest** is being attached to 3 different reports

<img width="837" alt="Screenshot 2024-09-06 at 10 30 55 AM" src="https://github.com/user-attachments/assets/bf954c15-5ec3-4e15-b122-8cc1303b5368">

hi @hsheth2  Let me your thought on this?